### DISCUSSION TITLE
새로운 branch

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#8] [Q&A] [answered]
새로운 branch를 만들어서 win32 클론 코딩 한 것을 넣어놓으려 합니다.
main아래에 자동으로 들어가는 것 같은데 제 코드들이 main에 종속되는 것이라서 다른 코드들과 섞이는 것 아닌가요?

새로운 branch를 제 깃헙에 추가할지 여기 cpp-stduy에 넣어야 할지 모르겠습니다.

+cpp-study는 왜 main에서 분기되나요

**Comment by Woon-2** [comment, marked as answer]
`main`은 편하게 `root branch`라고 생각하셔도 됩니다.

**`branch`는 분기라는 뜻이고, `merge`는 병합이라는 뜻이죠.**

"코드가 섞이는 것"은 서로 다른 `branch`들을 `merge`할 때만 일어납니다.

`merge`로 인해,
작업 공간이 복잡해질 것 같으면 폴더 정리를,
소스 파일 간 `conflict`가 예상되면 코드 수정을 해야겠죠.

새로운 `branch`를 만들면 최초에 기존 `branch`의 스냅샷을 갖고 있지만, 그 이후 변경 사항들은 기존 `branch`와 독립적입니다.

![capture_stepup1_2_1](https://user-images.githubusercontent.com/73771162/191155702-c7c0a093-24ea-4c52-b4e7-14730360c5f9.png)
(이미지 출처: [\[누구나 쉽게 이해할 수 있는 Git 입문\] 브랜치 만들기](https://backlog.com/git-tutorial/kr/stepup/stepup1_2.html))

물론 모든 프로젝트들은 최종 단계에서 `main`으로 모든 `branch`를 `merge`합니다.
그러나 아직 최종을 바라보기는 먼 단계이기도 하니, 신경 쓰지 않아주셨으면 하네요.

___

`Github Desktop`에서 `current branch`가 `main`인 상태에서 `create branch`를 하면 `main`에서 분기되는 것이고,
`current Branch`가 `Jongjin's-studyroom`인 상태에서 `create branch`를 하면 `Jongjin's-studyroom`에서 분기되지 않나요??
저도 확인 못 했으니 한번 확인해주시기 바랍니다.

___

추가적으로, 지금은 무엇이든 이 레포지터리에 올리는 것을 장려합니다.
win32 프로그래밍은 cpp-study라고 이름 붙은 이 레포지터리의 성격과는 맞지 않는 부분이 있죠.
지금은 그래도 여기에 올려주셨으면 합니다.

개인적으로 레포지터리를 만드셔서, 그곳에 업로드를 하신다면 상관 없습니다.
작성자분의 포트폴리오 작성에는 그 편이 좋아요.

개인 레포지터리에도 올리고, 이 레포지터리의 브랜치에도 올리셔도 괜찮습니다.

아예 이것이 새 프로젝트의 기획이 되어도 좋습니다.
win32 프로젝트도 해보자고 할 수도 있는 거니까요.

마음 가는대로 하시면 좋을 것 같습니다.
